### PR TITLE
fix for screenshots larger than 16,384 x 16,384

### PR DESCRIPTION
**Pull request by xSparfuchs**
dynamically adjusts chunkSizePixels to fit all chunks in 16,384 x 16,384 pixels

### CODE REVIEW

**Comment by tobspr** [comment]
I think you mixed tabs an spaces :)

**Comment by xSparfuchs** [comment]
looks like i did an oopsie there, should be fine now

File: src/js/game/hud/parts/screenshot_exporter.js [hunk 1]
```diff
@@ -56,7 +56,17 @@ export class HUDScreenshotExporter extends BaseHUDPart {
         const dimensions = maxChunk.sub(minChunk);
         logger.log("Dimensions:", dimensions);
 
-        const chunkSizePixels = 128;
+        let chunkSizePixels = 128;
+
+        if (dimensions.x + dimensions.y > 256) {
+            if (dimensions.x > dimensions.y) {
+                chunkSizePixels = Math.floor(128 * (128 / dimensions.x));
```
Comment (tobspr) [inline src/js/game/hud/parts/screenshot_exporter.js:63]: Why not:

`chunkSizePixels = Math.floor(128 * (128 / Math.max(dimensions.x, dimensions.y)));` 

Also I think it'd be simpler to do:

`const maxDimensions = Math.max(dimensions.x, dimensions.y)`;

Then:

`if (maxDimensions > 128)`

And then divide by `maxDimensions`

Comment (xSparfuchs) [reply src/js/game/hud/parts/screenshot_exporter.js]: yup, that makes a lot of sense

**Comment by tobspr** [comment]
Thank you!

**Comment by tobspr** [comment]
Will be in 1.1.16

**Pull request merged**